doc/connectivity/networking/api/lwm2m.rst

```diff
@@ -364,14 +364,43 @@ endpoint name.  This is important as it needs to be unique per LwM2M server:
 	(void)memset(&client, 0x0, sizeof(client));
 	lwm2m_rd_client_start(&client, "unique-endpoint-name", 0, rd_client_event);
 
-Using LwM2M library with DTLS
-*****************************
+.. _lwm2m_security:
 
-The Zephyr LwM2M library can be used with DTLS transport for secure
-communication by selecting :kconfig:option:`CONFIG_LWM2M_DTLS_SUPPORT`.  In the client
-initialization we need to create a PSK and identity.  These need to match
-the security information loaded onto the LwM2M server.  Normally, the
-endpoint name is used to lookup the related security information:
+LwM2M security modes
+********************
+
+The Zephyr LwM2M library can be used either without security or use DTLS to secure the communication channel.
+When using DTLS with LwM2M engine, PSK (Pre-Shared Key) and X.509 certificates are two security modes that can be used to secure the communication.
+The engine uses LwM2M Security object (Id 0) to read stored credentials and feed keys from security object into
+the TLS credential subsystem, see :ref:`secure sockets documentation <secure_sockets_interface>`.
+When security is used :kconfig:option:`CONFIG_LWM2M_DTLS_SUPPORT` Kconfig option must be enabled.
+
+Depending on the selected mode, security object should contain following data:
+
+PSK
+  Security Mode (Resource ID 2) should be set to zero (Pre-Shared Key mode).
+  Identity (Resource ID 3) contains PSK ID in binary form.
+  Secret key (Resource ID 5) contains the PSK key in binary form.
+  If key or identity is provided as a hex string, it must be converted to binary before storing into security object.
+
+X509
+  When X509 certificates are used, Security Mode (ID 2) should be set to 2 (Certificate mode).
+  Identity (ID 3) is used to store the client certificate and Secret key (ID 5) must have a private key associated with the certificate.
+  Server Public Key resource (ID 4) must contain a server certificate or CA certificate used to sign the certificate chain.
+  If :kconfig:option:`CONFIG_MBEDTLS_PEM_CERTIFICATE_FORMAT` is enabled, certificates and private key can be fed as a PEM format. Otherwise they must be in binary DER format.
```

Comment:
```suggestion
  If the :kconfig:option:`CONFIG_MBEDTLS_PEM_CERTIFICATE_FORMAT` Kconfig option is enabled, certificates and private key can be entered in PEM format.
  Otherwise, they must be in binary DER format.
```